File: src/ProjectR/content/scripts/spells/Hurricane.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Hurricane"
end

function GetDescription()
    return "Summon a hurricane to blast away your enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .28
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { WND }
end

function SpellEffect()
  local damage = (4.375 * aMD * (aWND/100) - .875 * dMR) * (100/dWND)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/Hurricane.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class Hurricane : SpellScriptBase
    {
        public override string Name { get { return "Hurricane"; } }
        public override string Description { get { return "Summon a hurricane to blast away your enemies."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.WND, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .28; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4.375 * cMD * (cWND / 100) - .875 * tMR) * (100 / tWND);
            DealDamage(damage);
        }
    }
}